tests/firedrake/regression/test_interpolate.py

```diff
@@ -566,3 +566,37 @@ def test_interpolate_logical_not():
     a = assemble(interpolate(conditional(Not(x < .2), 1, 0), V))
     b = assemble(interpolate(conditional(x >= .2, 1, 0), V))
     assert np.allclose(a.dat.data, b.dat.data)
+
+
+@pytest.mark.parametrize("rank", range(3))
+@pytest.mark.parametrize("dim", range(1, 4))
+def test_clement_interpolator_simplex(dim, rank):
```

Comment:
We should test a vector with 3 component in a 2D mesh.

Reply:
Good suggestion - refactoring for this helped to address some limitations in the implementation. Addressed in 7ec5197c3f525896e3ce51b9b9bc71dbfb146c00.